Lib/test/test_import/__init__.py

```diff
@@ -80,6 +80,43 @@ def test_from_import_missing_attr_raises_ImportError(self):
         with self.assertRaises(ImportError):
             from importlib import something_that_should_not_exist_anywhere
 
+    def test_from_import_missing_attr_has_name_and_path(self):
+        """
+        Make sure that from ... import ... raise an import error that has both
+        `name` and `path` set when both are available.
+        """
+
+        try:
+            from os import i_dont_exist
+        except ImportError as e:
+            self.assertEqual(e.name, 'os',
+                "ImportError.name is set")
```

Comment:
The custom message isn't necessary as what the test is checking is obvious (same goes for all of the other messages).